cishouseholds/merge.py

```diff
@@ -53,3 +53,44 @@ def join_dataframes(df1: DataFrame, df2: DataFrame, reference_column: str, join_
         Specify join type to apply to .join() method
     """
     return df1.join(df2, on=reference_column, how=join_type)
+
+
+def assign_merge_process_group_flag(
+    df: DataFrame,
+    column_name_to_assign: str,
+    out_of_date_range_flag: str,
+    count_barcode_labs_column_name: str,
+    count_barcode_labs_condition: str,
+    count_barcode_voyager_column_name: str,
+    count_barcode_voyager_condition: str,
+):
+    """
+    Combine three conditions to create flag indicating record to be processed in forthcoming matching process.
```

Comment:
This is run for each of 1:1, 1:many, many:1 and many:many to identify the relevant processing group.